lib/bom.js

```diff
@@ -368,6 +368,11 @@ declare class CloseEvent extends Event {
     wasClean: boolean;
 }
 
+type MessageEventHandler = (event: MessageEvent) => mixed;
+type CloseEventHandler = (event: CloseEvent) => mixed;
+type MessageEventListener = {handleEvent: MessageEventHandler} | MessageEventHandler;
```

Comment:
@gabelevi that's a bummer. I can give this a spin re: performance. Did any optimizations ever happen since #2680?